src/components/ParseProgress.tsx

```
@@ -0,0 +1,33 @@
+  progress: number;
+  isVisible: boolean;
+}
+
+/**
+ * Progress indicator for chat file parsing
+ */
+export const ParseProgress = ({ progress, isVisible }: ParseProgressProps) => {
+  if (!isVisible) return null;
+
+  logger.debug('⏳ [COMP] ParseProgress render, progress:', progress);
+  useEffect(() => {
```

ESLint

[error] React Hook "useEffect" is called conditionally. React Hooks must be called in the exact same order in every component render. Did you accidentally call a React Hook after an early return?

(react-hooks/rules-of-hooks)